extensions/panache/hibernate-orm-rest-data-panache/deployment/src/main/java/io/quarkus/hibernate/orm/rest/data/panache/deployment/AbstractDataAccessImplementor.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.hibernate.orm.rest.data.panache.deployment;
+
+import static io.quarkus.gizmo.MethodDescriptor.ofMethod;
+
+import io.quarkus.gizmo.BranchResult;
+import io.quarkus.gizmo.BytecodeCreator;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.rest.data.panache.deployment.utils.CollectionImplementor;
+import io.quarkus.rest.data.panache.deployment.utils.ReflectionImplementor;
+
+public abstract class AbstractDataAccessImplementor {
+
+    private final ReflectionImplementor reflectionImplementor = new ReflectionImplementor();
+
+    private final CollectionImplementor collectionImplementor = new CollectionImplementor();
+
+    protected void specificFieldUpdate(BytecodeCreator creator, ResultHandle foundEntity,
+            ResultHandle inputEntity) {
+        ResultHandle listOfFields = reflectionImplementor.getFieldsAsList(creator, inputEntity);
```

Comment:
Yes, that is absolutely correct.

We only want reflection as a last resort

Reply:
@gytis sure, I'll try to look in that direction and provide another solution asap

Reply:
I've done something similar just with a getter in this PR https://github.com/quarkusio/quarkus/pull/14249/files#diff-52579ac7c4d5c6198a4e6ae7892dcbd8e6ed5f753854f3685d72f47d27561ca3R17. So feel free to take a look there, it might be helpful.

Reply:
@gytis thanks!